### DISCUSSION TITLE
What exactly is taps?

### DISCUSSION THREAD

**Discussion by yuqilin1988** [risc0/risc0#729] [Q&A] [answered]
Hi everyone, I'm a newcomer. What exactly is taps? The comment says that taps are automatically generated. What should I do to generate this code?

**Comment by pdg744** [comment, marked as answer]
A `tap` is a tuple `(n,t)` that serves as a (relative) reference to a location in the execution trace. 
Here `n` identifies a column in the trace and `t` describes how many steps forward/backward to look.  
Full details are here: https://dev.risczero.com/proof-system-in-detail.pdf

We don't currently support an option for users to generate this code.
